### ISSUE TITLE
No alpaka::dev::cpu::detail::cpu for PGI

### ISSUE THREAD

**Issue by jkelling** [alpaka-group/alpaka#1117]
The guards for how the `cpuid()` function should be defined does not consider all existing compilers and has no `#else`, see
https://github.com/alpaka-group/alpaka/blob/6310424eff951eddedd0af8d26b8aa0012254075/include/alpaka/dev/cpu/SysInfo.hpp#L50

I suggest to reformulate the checks to turn the first branch into `#else`.

However, I am not sure how to do that, because I do not understand what the condition
`(!BOOST_COMP_MSVC_EMULATED && defined(__INTEL_COMPILER))`
is supposed to mean. Should we not always not have msvc if we are compiling with Intel?

**Comment by BenjaminW3** [comment]
This is more of a platform detection than a compiler detection. On Windows the Intel compiler says that it emulates the Microsoft compiler but it should still use the windows version but on Linux the Intel compiler should use the Linux version. There may be better ways to express this.

**Comment by jkelling** [comment]
Maybe then the OS should be detected:
```C++
#if BOOST_OS_UNIX || BOOST_OS_MACOS || BOOST_OS_CYGWIN // I guess ...
...
#elif BOOST_OS_WINDOWS
...
#else
void cpuid (...)
{
  asm( ... )
  //or
  ex[0] = ex[1] = ex[2] = ex[3] = 0;
}
#endif
```

**Comment by jkelling** [comment]
`cpuid.h` is provided by GCC, not unix, so detecting the compiler is correct. PGI is using the GCC STL and also gets access to GCC's `cpuid.h`, so it it could be added to the list.

I still think there should be a `#else` to catch compilers we did not think of.

**Comment by jkelling** [comment]
I made a PR for this: https://github.com/alpaka-group/alpaka/pull/1119 .

Apparently I am not allowed to link it to the issue, can someone please link them?

**Comment by sbastrakov** [comment]
Solved with #1119, closing.
